src/v/redpanda/debug_bundle.cc

```diff
@@ -0,0 +1,266 @@
+}
+
+debug_bundle::debug_bundle(ss::sstring filename, ss::sstring rpk_path)
+  : _status{debug_bundle_status::initialized}
+  , _filename{filename}
+  , _rpk_cmd{make_rpk_cmd(rpk_path)}
+  , _transfer_handle{_filename, nullptr, false} {}
+
+ss::future<> debug_bundle::stop() {
+    vlog(bundle_log.info, "Stopping debug bundle ...");
+    return ss::make_ready_future<>();
+}
+
+ss::future<> debug_bundle::start_creating_bundle(
+  const request_auth_result& auth_state, const debug_bundle_params params) {
+    if (ss::this_shard_id() != debug_bundle_shard_id) {
+        return container().invoke_on(
+          debug_bundle_shard_id,
+          [&auth_state, params{std::move(params)}](debug_bundle& b) {
+              return b.start_creating_bundle(auth_state, std::move(params));
+          });
+    }
+
+    ss::sstring host_path;
+    auto host_env = std::getenv("PATH");
+    if (!host_env) {
+        vlog(
+          bundle_log.warn,
+          "Failed to get 'PATH' environmental variable, the debug bundle may "
+          "be incomplete due to missing dependencies");
+    } else {
+        vlog(bundle_log.trace, "Host path: {}", host_env);
+        host_path = fmt::format("PATH={}", host_env);
+    }
+
+    std::vector<ss::sstring> rpk_argv{
+      _rpk_cmd, "debug", "bundle", "--output", _filename};
+    // Add SASL creds to RPK flags if SASL is enabled.
+    // No need to check for sasl on the broker endpoint because that is for
+    // Kafka brokers where as the bundle is managed by the Admin server only.
+    if (config::shard_local_cfg().enable_sasl()) {
+        rpk_argv.push_back("--user");
+        rpk_argv.push_back(auth_state.get_username());
+        rpk_argv.push_back("--password");
+        rpk_argv.push_back(auth_state.get_password());
+        rpk_argv.push_back("--sasl-mechanism");
+        rpk_argv.push_back(auth_state.get_mechanism());
+    }
+
+    if (params.logs_since.has_value()) {
+        rpk_argv.push_back("--logs-since");
+        rpk_argv.push_back(params.logs_since.value());
+    }
+
+    if (params.logs_until.has_value()) {
+        rpk_argv.push_back("--logs-until");
+        rpk_argv.push_back(params.logs_until.value());
+    }
+
+    if (params.logs_size_limit.has_value()) {
+        rpk_argv.push_back("--logs-size-limit");
+        rpk_argv.push_back(params.logs_size_limit.value());
+    }
+
+    if (params.metrics_interval.has_value()) {
+        rpk_argv.push_back("--metrics-interval");
+        rpk_argv.push_back(params.metrics_interval.value());
+    }
+
+    vlog(
+      bundle_log.trace, "RPK absolute path {}, flags {}", _rpk_cmd, rpk_argv);
+    ssx::background
+      = ss::experimental::spawn_process(
+          _rpk_cmd.c_str(), {.argv = std::move(rpk_argv), .env = {host_path}})
+          .then([this](auto process) {
+              auto stdout = process.stdout();
+              _status = debug_bundle_status::running;
+              return ss::do_with(
+                std::move(process),
```

Comment:
Since we're already moving process into the `do_with` do you need to also move `stdout` which comes from `process`?

Reply:
I am confused by this comment, `stdout` is already moved into the `do_with` on the next line.

```cpp
std::move(stdout), // <-- moved here
[](auto& p, auto& stdout) { // <-- captured here
```

Reply:
Spoke offline with @michael-redpanda:
Scylla did this in their test https://github.com/scylladb/seastar/blob/fdbe8954afcfdde8e01c8a14a8d3c0e39950d9d2/tests/unit/spawn_test.cc#L67 so I figured we should do the same.